contracts/TokenConverter/RiskFundConverter.sol

```diff
@@ -0,0 +1,398 @@
+                    }
+                }
+                unchecked {
+                    ++i;
+                }
+            }
+            assetsReserves[tokenAddress] -= amountDiff;
+        }
+    }
+
+    /// @notice Update the poolAssetsReserves upon transferring the tokens
+    /// @param pool Address of the pool
+    /// @param tokenAddress Address of the token
+    /// @param amount Amount transferred to address(to)
+    /// @param assetReserve Asset's reserve for the pool
+    /// @custom:event AssetsReservesUpdated emits on success
+    function updatePoolAssetsReserve(
+        address pool,
+        address tokenAddress,
+        uint256 amount,
+        uint256 assetReserve
+    ) internal returns (uint256 poolAmountShare) {
+        poolAmountShare = (poolsAssetsReserves[pool][tokenAddress] * amount) / assetReserve;
+        poolsAssetsReserves[pool][tokenAddress] -= poolAmountShare;
+        emit AssetsReservesUpdated(pool, tokenAddress, poolAmountShare);
+    }
+
+    /// @notice Update the poolsAssetsDirectTransfer mapping
+    /// @param comptrollers Addresses of the pools
+    /// @param assets Addresses of the assets need to be added for direct transfer
+    /// @param values Boolean value to indicate whether direct transfer is allowed for each asset.
+    /// @custom:event PoolAssetsDirectTransferUpdated emits on success
+    /// @custom:error InvalidArguments thrown when comptrollers array length is not equal to assets array length
+    function _setPoolsAssetsDirectTransfer(
+        address[] calldata comptrollers,
+        address[][] calldata assets,
+        bool[][] calldata values
+    ) internal {
+        uint256 comptrollersLength = comptrollers.length;
+
+        if ((comptrollersLength != assets.length) || (comptrollersLength != values.length)) {
+            revert InvalidArguments();
+        }
+
+        for (uint256 i; i < comptrollersLength; ) {
+            address[] memory poolAssets = assets[i];
+            bool[] memory assetsValues = values[i];
+            uint256 poolAssetsLength = poolAssets.length;
+
+            if (poolAssetsLength != assetsValues.length) {
+                revert InvalidArguments();
+            }
+
+            for (uint256 j; j < poolAssetsLength; ) {
+                poolsAssetsDirectTransfer[comptrollers[i]][poolAssets[j]] = assetsValues[j];
+                emit PoolAssetsDirectTransferUpdated(comptrollers[i], poolAssets[j], assetsValues[j]);
+                unchecked {
+                    ++j;
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// @notice This function checks for the given asset is listed in core pool or not
+    /// @param tokenAddress Address of the asset
+    /// @return isAssetListed true if the asset is listed
+    function isAssetListedInCore(address tokenAddress) internal view returns (bool isAssetListed) {
```

Comment:
I wonder if we want to also support VAI here. Wdyt?

Reply:
VAI fees are sent directly to the VTreasury. I think the only way to have here VAI would be if we have a market for VAI (in the Core pool or in one IL market), and in that case, I think the current code covers the case